File: src/main/java/de/dhbw/mannheim/database/entities/Message.java
package de.dhbw.mannheim.database.entities;

import javax.persistence.*;

@Entity
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    private String content;

    @ManyToOne
    private Citizen citizen;

    @ManyToOne
    private Process process;

    @ManyToOne
    private Authority authority;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Citizen getCitizen() {
        return citizen;
    }

    public void setCitizen(Citizen citizen) {
        this.citizen = citizen;
    }

    public Process getProcess() {
        return process;
    }

    public void setProcess(Process process) {
        this.process = process;
    }

    public Authority getAuthority() {
        return authority;
    }

    public void setAuthority(Authority authority) {
        this.authority = authority;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: Frontend/src/app/Entities/Message.ts
import {Citizen} from "./Citizen";
import {Process} from "./Process";
import {Authority} from "./Authority";

export class Message{

  id: number;
  content: string;
  citizen: Citizen;
  process: Process;
  authority: Authority;

  constructor(id: number,content: string,citizen: Citizen,process: Process,authority: Authority) {
    this.id = id;
    this.content = content;
    this.citizen = citizen;
    this.process= process;
    this.authority = authority;
  }
}
